File: #0062.unique-paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {

        vector<vector<int>> up(m + 1, vector(n + 1, 0));
        up[1][1] = 1;
        for (int i = 1; i <= m; i++) {
            for (int j = 1; j <= n; j++) {
                if (i == 1 && j == 1) {
                    continue;
                }
                up[i][j] = up[i-1][j] + up[i][j-1];
            }
        }
        return up[m][n];
    }
};

File: #0062.unique-paths.rs
impl Solution {
    pub fn unique_paths(m: i32, n: i32) -> i32 {

        let (m, n) = (m as usize, n as usize);
        let mut up = vec![1; n];
        for i in 1..m {
            for j in 1..n {
                up[j] += up[j - 1];
            }
        }
        return up[n - 1];
    }
}
